lib/src/r_tree/leaf_node.dart

```diff
@@ -22,7 +22,20 @@ class LeafNode<E> extends Node<E> {
   List<RTreeDatum<E>> _items = [];
   List<RTreeDatum<E>> get children => _items;
 
-  LeafNode(int branchFactor) : super(branchFactor);
+  LeafNode(int branchFactor, {List<RTreeDatum<E>>? initialItems})
+      : super(branchFactor) {
+    if (initialItems != null) {
+      if (initialItems.length > branchFactor) {
+        throw ArgumentError.value('too many items');
+      }
+      _items = initialItems;
+
+      updateBoundingRect();
+    }
+  }
```

Comment:
Can we make this non-nullable?
```suggestion
  LeafNode(int branchFactor, {List<RTreeDatum<E>> initialItems = const []})
      : super(branchFactor) {
    if (initialItems.isEmpty) {
      return;
    }
    if (initialItems.length > branchFactor) {
      throw ArgumentError.value('too many items');
    }
    _items = initialItems;

    updateBoundingRect();
  }
```

Reply:
That actually introduces a type error since `const []` isn't of type `List<RTreeDatum<E>>`, and I can't use that type in the initializer. Unless I'm missing something, I don't think I can make it non-nullable.